src/graphql/graphql-api.php

```diff
@@ -0,0 +1,245 @@
+<?php
+/**
+ * GraphQL API
+ *
+ * @package vip-block-data-api
+ */
+
+namespace WPCOMVIP\BlockDataApi;
+
+defined( 'ABSPATH' ) || die();
+
+/**
+ * GraphQL API to offer an alternative to the REST API.
+ */
+class GraphQLApi {
+	/**
+	 * Initiatilize the graphQL API, if its allowed
+	 *
+	 * @access private
+	 */
+	public static function init() {
+		/**
+		 * Filter to enable/disable the graphQL API. By default, it is enabled.
+		 * 
+		 * @param bool $is_graphql_to_be_enabled Whether the graphQL API should be enabled or not.
+		 */
+		$is_graphql_to_be_enabled = apply_filters( 'vip_block_data_api__is_graphql_enabled', true );
+
+		if ( ! $is_graphql_to_be_enabled ) {
+			return;
+		}
+
+		add_filter( 'vip_block_data_api__sourced_block_result', [ __CLASS__, 'transform_block_format' ], 10, 5 );
+
+		add_action( 'graphql_register_types', [ __CLASS__, 'register_types' ] );
+	}
+
+	/**
+	 * Extract the blocks data for a post, and return back in the format expected by the graphQL API.
+	 *
+	 * @param  WPGraphQL\Model\Post $post_model Post model for post.
+	 * @return array
+	 */
+	public static function get_blocks_data( $post_model ) {
```

Comment:
Is there a PHP version policy for this plugin? Can you use type declarations?
```suggestion
	public static function get_blocks_data( \WPGraphQL\Model\Post $post_model ) {
```

Reply:
We have set it at 7.4, but will be upping it to 8.0. At that point we were gonna go in and set the type declarations. That is on the cards, so I haven't done that here.

That said, I do have to figure out when I do how to pass in post model of type \WPGraphQL\Model\Post when we dont bundle in WP-GraphQL with the plugins. The tests fail due to it, I realized.

I'll leave this one as is for this PR